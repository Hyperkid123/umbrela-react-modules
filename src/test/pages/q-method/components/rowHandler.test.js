import React from 'react'
import RowHandler from '../../../../pages/q-method/components/rowHandler';
import {shallow, mount} from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Pyramid row handler', () => {
  it('should render correclty' , () => {
      const tree = shallow(
        <RowHandler rows={2} addRow={jest.fn()} removeRow={jest.fn()}/>
      ).dive();
      expect(toJson(tree)).toMatchSnapshot();
  });

  it('should call functions on click', () => {
    const changeRows = jest.fn();
    const renderer = shallow(<RowHandler rows={5} addRow={changeRows} removeRow={changeRows}/>);
    // NOTE: mui-broke full dom renderinf, fix after mui update
    //renderer.find('#increase-blocks-0').simulate('click');
    //expect(changeTiles.mock.calls.length).toEqual(1);
  })
});
