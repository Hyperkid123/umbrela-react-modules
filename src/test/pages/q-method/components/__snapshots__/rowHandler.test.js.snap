// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pyramid row handler should render correclty 1`] = `
<div
  className="sc-bdVaJa fizmbb"
>
  <styled.div
    horizontalCenter={true}
  >
    <styled.div>
      <WithStyles
        title="Přidat řádedek"
      >
        <WithStyles
          onClick={[Function]}
        >
          <ShouldUpdate />
        </WithStyles>
      </WithStyles>
      <WithStyles
        title="Odebrat poslední řádek"
      >
        <div>
          <WithStyles
            disabled={false}
            onClick={[Function]}
          >
            <ShouldUpdate />
          </WithStyles>
        </div>
      </WithStyles>
    </styled.div>
  </styled.div>
  <styled.div />
</div>
`;
